packages/select/src/Select/Select.tsx

```diff
@@ -121,7 +123,11 @@ export const Select = forwardRef<HTMLDivElement, SelectProps>(
     const descriptionId = `${id}-description`;
     const menuId = `${id}-menu`;
 
-    const [open, setOpen] = useState(false);
+    const [uncontrolledOpen, setUncontrolledOpen] = useState(false);
+
+    // Use controlled open state if provided, otherwise use internal state
+    const open = openProp !== undefined ? openProp : uncontrolledOpen;
+    const setOpen = openProp !== undefined ? setOpenProp! : setUncontrolledOpen;
```

Comment:
This still isn't correct though, since it'll use the internal setter with the external state value
```suggestion
    const open = isDefined(openProp) ? openProp : uncontrolledOpen;
    const setOpen = isDefined(openProp) && isDefined(setOpenProp) ? setOpenProp : setUncontrolledOpen;
```